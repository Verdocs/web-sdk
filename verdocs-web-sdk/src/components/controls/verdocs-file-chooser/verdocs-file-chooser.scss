@import '../../../theme.scss';

verdocs-file-chooser {
  display: flex;
  flex-direction: column;
  background-color: $verdocs-grey-4;
  font-family: $verdocs-primary-font;

  .upload-box {
    text-align: center;
    box-sizing: border-box;
    padding: 44px 18px 66px;
    color: rgba(0, 0, 0, 0.54);

    svg {
      width: 64px;
      height: 64px;
      fill: #5c6575;
    }
  }

  .selected-filename {
    font-size: 20px;
    margin-top: 20px;
    font-weight: bold;
    overflow-wrap: anywhere;
  }

  .or-prefer {
    height: 20px;
    margin: 20px 0;
    font-size: 16px;
  }
}
