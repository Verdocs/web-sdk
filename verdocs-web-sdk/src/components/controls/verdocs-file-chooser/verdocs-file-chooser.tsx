import {VerdocsEndpoint} from '@verdocs/js-sdk';
import {Component, h, Event, EventEmitter, Prop, State, Host} from '@stencil/core';
import {unicodeNBSP} from '../../../utils/Icons';

/**
 * Displays a file picker to upload an attachment. This component is just the picker - the host application or component should
 * provide the actual upload functionality.
 */
@Component({
  tag: 'verdocs-file-chooser',
  styleUrl: 'verdocs-file-chooser.scss',
  shadow: false,
})
export class VerdocsFileChooser {
  /**
   * The endpoint to use to communicate with Verdocs. If not set, the default endpoint will be used.
   */
  @Prop() endpoint: VerdocsEndpoint = VerdocsEndpoint.getDefault();

  /**
   * Event fired when a file has been selected. Note that the file may be null if the user is choosing a different file.
   * Host applications should use this event to enable/disable buttons to upload or otherwise process the selected file.
   */
  @Event({composed: true}) fileSelected: EventEmitter<{file: File | null}>;

  @State() file: File | null;

  handleFileChanged(e: any) {
    this.file = e.target.files?.[0] || null;
    this.fileSelected?.emit({file: this.file});
    console.debug('[CHOOSER] Selected file', this.file);
  }

  handleSelectFile(e: any) {
    e.stopPropagation();
    this.file = null;
    this.fileSelected?.emit({file: null});
    document.getElementById('verdocs-file-chooser')?.click();
  }

  render() {
    const buttonlabel = this.file ? 'Select a different file' : 'Select a file from your computer';

    return (
      <Host>
        <input
          type="file"
          multiple
          id="verdocs-file-chooser"
          accept="application/pdf,.pdf,.doc,.docx,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document"
          style={{display: 'none'}}
          onChange={e => this.handleFileChanged(e)}
        />

        <div class="upload-box">
          {/*<div innerHTML={FileIcon} />*/}

          <div class="selected-filename">{this.file ? this.file.name : 'Drag a file here'}</div>

          <div class="or-prefer">{this.file ? unicodeNBSP : 'Or, if you prefer...'}</div>

          <verdocs-button label={buttonlabel} size="small" onClick={e => this.handleSelectFile(e)} />
        </div>
      </Host>
    );
  }
}
