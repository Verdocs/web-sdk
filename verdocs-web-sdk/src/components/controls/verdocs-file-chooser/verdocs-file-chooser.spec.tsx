import { newSpecPage } from '@stencil/core/testing';
import { VerdocsFileChooser } from './verdocs-file-chooser';

describe('verdocs-file-chooser', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsFileChooser],
      html: `<verdocs-file-chooser></verdocs-file-chooser>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with a custom endpoint', async () => {
    const page = await newSpecPage({
      components: [VerdocsFileChooser],
      html: `<verdocs-file-chooser endpoint="custom"></ververdocs-file-chooser>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('emits fileSelected event on file input change', async () => {
    const page = await newSpecPage({
      components: [VerdocsFileChooser],
      html: `<verdocs-file-chooser></verdocs-file-chooser>`,
    });
    const input = page.root.querySelector('input[type="file"]');
    const spy = jest.fn();
    page.root.addEventListener('fileSelected', spy);

    // Simulate file selection
    const file = new File(['dummy content'], 'test.pdf', { type: 'application/pdf' });
    const event = {
      target: { files: [file] }
    };
    (page.rootInstance as any).handleFileChanged(event);
    await page.waitForChanges();

    expect(spy).toHaveBeenCalled();
    expect((page.rootInstance as any).file).toBe(file);
  });
});
