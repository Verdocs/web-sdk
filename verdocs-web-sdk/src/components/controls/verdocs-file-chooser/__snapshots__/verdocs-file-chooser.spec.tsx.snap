// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-file-chooser renders with a custom endpoint 1`] = `
<verdocs-file-chooser endpoint="custom">
  <input accept="application/pdf,.pdf,.doc,.docx,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document" id="verdocs-file-chooser" multiple="" type="file" style="display: none;">
  <div class="upload-box">
    <div class="selected-filename">
      Drag a file here
    </div>
    <div class="or-prefer">
      Or, if you prefer...
    </div>
    <verdocs-button label="Select a file from your computer" size="small"></verdocs-button>
  </div>
</verdocs-file-chooser>
`;

exports[`verdocs-file-chooser renders with default props 1`] = `
<verdocs-file-chooser>
  <input accept="application/pdf,.pdf,.doc,.docx,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document" id="verdocs-file-chooser" multiple="" type="file" style="display: none;">
  <div class="upload-box">
    <div class="selected-filename">
      Drag a file here
    </div>
    <div class="or-prefer">
      Or, if you prefer...
    </div>
    <verdocs-button label="Select a file from your computer" size="small"></verdocs-button>
  </div>
</verdocs-file-chooser>
`;
